docs/design/inference.md

```diff
@@ -1,15 +1,52 @@
 # Inference
 
-Notes about inference in MXFusion.
+## Overview
+
+Inference in MXFusion is broken down into a few logical pieces that can be combined together as necessary.
+
+The highest level object you'll deal with will be derived from the ```mxfusion.inference.Inference``` class. This is the outer loop that drives the inference algorithm, holds the relevant parameters and models for training, and handles serialization after training. At a minimum, ```Inference``` objects take as input the ```InferenceAlgorithm``` to run and the model it will be run on. On creation, an ```InferenceParameters``` object is created and attached to the ```Inference``` method which will store and manage (MXNet) parameters during inference.
```

Comment:
The creation of an ```Inference``` object does not directly require the model object.

```suggestion
The highest level object you'll deal with will be derived from the ```mxfusion.inference.Inference``` class. This is the outer loop that drives the inference algorithm, holds the relevant parameters and models for training, and handles serialization after training. At a minimum, ```Inference``` objects take as input the ```InferenceAlgorithm``` to run. On creation, an ```InferenceParameters``` object is created and attached to the ```Inference``` method which will store and manage (MXNet) parameters during inference.
```